File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/diagnosiskeydist/apispec/ZIPSubmissionPeriod.java
package uk.nhs.nhsx.diagnosiskeydist.apispec;

import java.time.Duration;
import java.time.Instant;
import java.util.List;

public interface ZIPSubmissionPeriod {
    String zipPath();

    boolean isCoveringSubmissionDate(Instant diagnosisKeySubmission, Duration periodOffset); //for the server for calculation

    List<? extends ZIPSubmissionPeriod> allPeriodsToGenerate();

    Instant getEndExclusive(); // < end from client perspective -> url

    Instant getStartInclusive(); // >= start from client perspective
}

File: src/aws/lambdas/incremental_distribution/cta/src/main/java/uk/nhs/nhsx/diagnosiskeydist/apispec/ZIPSubmissionPeriod.kt
package uk.nhs.nhsx.diagnosiskeydist.apispec

import java.time.Duration
import java.time.Instant

interface ZIPSubmissionPeriod {
    fun zipPath(): String
    fun isCoveringSubmissionDate(diagnosisKeySubmission: Instant, periodOffset: Duration): Boolean
    fun allPeriodsToGenerate(): List<ZIPSubmissionPeriod>
    val endExclusive: Instant
    val startInclusive: Instant
}
